### ISSUE TITLE
Add support for an optinal type guard for the query functions

### ISSUE THREAD

**Issue by googol** [sluukkonen/possu#16]
For example for `query` something like:

```ts
export async function query<T extends object = unknown>(
  client: Pool | PoolClient,
  sql: SqlQuery,
  guard: (value: unknown): value is T = (value) => true
): Promise<T[]> {
```

So by default you'd get `unknown` as `T`, but you could change that by giving a guard function. You get these guards for free when using runtypes or io-ts, so it shouldn't be too much of a burden.

If performance is an issue, the use of the guard func to validate the query result could be dependent on NODE_ENV for example

**Comment by sluukkonen** [comment]
I was thinking of something similar today, although I think I'd rather make them separate functions.

Maybe we could have `query(client, sql) = queryWith(identity, client, sql)`, `queryOne(parser, client, sql) === queryOneWith(identity, client, sql)` and so on.

**Comment by googol** [comment]
That could work too. My main concern though is to have the library not let me shoot myself in the foot by allowing unchecked type coercion or `any` types coming out of these kinds of functions.

So IMO the return type should be `unkown` (`unknown[]` for the ones returning multiple elements) and only allow changing the type by somehow validating what is returned

**Comment by sluukkonen** [comment]
@googol I tried to tackle this in https://github.com/sluukkonen/possu/pull/17

Let me know what you think.

**Comment by sluukkonen** [comment]
Fixed by #17
